bash/utils.sh

```diff
@@ -252,10 +252,24 @@ function compress {
 }
 
 
+# taken from https://unix.stackexchange.com/a/22215
+function find-up {
+    path=$(pwd)
+    while [[ "$path" != "" && ! -e "$path/$1" ]]; do
+        path=${path%/*}
+    done
+    echo "$path"
+}
+
+
 function upload {
     local dataset_name=${1}
     local subfolder=${2} # typically version
-    local branchname=$(git rev-parse --symbolic-full-name --abbrev-ref HEAD)
+    local root_folder=$(find-up .git)
+    local wd=$(pwd)
+    cd root_folder
+    local branchname=$(cat .git/HEAD | cut -d '/' -f3)
```

Comment:
Removed dependency on git for checking branch! This would have solved the devdb issue as well.